python/utils/LinkedLibraryHolder.cpp

```diff
@@ -154,26 +154,18 @@ LinkedLibraryHolder::LinkedLibraryHolder() {
       auto idx = simName.find_last_of(".");
       simName = simName.substr(0, idx);
 
-      // FIXME until we have a better handle on MPI init / finalize
-      // we can't load these
-      if (simName == "tensornet" || simName == "cuquantum_mgpu") {
-        simulators.emplace(simName, nullptr);
-        continue;
-      }
-
       // Store the dlopen handles
       auto iter = libHandles.find(path.string());
       if (iter == libHandles.end())
         libHandles.emplace(path.string(), dlopen(path.string().c_str(),
                                                  RTLD_GLOBAL | RTLD_NOW));
 
       // Load the plugin and get the CircuitSimulator.
-      std::string symbolName = fmt::format("getCircuitSimulator_{}", simName);
-      auto *simulator =
-          getUniquePluginInstance<nvqir::CircuitSimulator>(symbolName);
-
       cudaq::info("Found simulator plugin {}.", simName);
-      simulators.emplace(simName, simulator);
+      simulators.emplace(simName, [simName]() {
+        return getUniquePluginInstance<nvqir::CircuitSimulator>(
+            fmt::format("getCircuitSimulator_{}", simName));
```

Comment:
Q: can't this just be a simple append string operation? (`fmt::format` seems kind of heavyweight for that.)

Reply:
This can be updated to just a string + op